### DISCUSSION TITLE
Strange design changes in new Version 58.21

### DISCUSSION THREAD

**Discussion by mcliquid** [Helium314/SCEE#600] [General]
@Helium314 
Compared to the previous versions, the settings look completely wrong. The spaces between the lines are much too large, the font is very different and mostly too small. Many translations of labels are missing.
Is this part of the changeover to iOS (Multi Platform) and is it intentional or has something gone wrong here?

![quests](https://github.com/user-attachments/assets/64c9cd7e-fc5b-4c2c-8e75-9396e0868b67)
![settings](https://github.com/user-attachments/assets/5ec1c0f2-d8f8-49ca-870b-fe7b55f25d1a)

**Comment by Helium314** [comment]
> Many translations of labels are missing

Well, that's how it is when e.g. `questList_disabled_by_default` does not exist in the [German `strings.xml`](https://github.com/Helium314/SCEE/blob/modified/app/src/main/res/values-de/strings.xml). I think SC translations are pulled only before SC version update.

> the settings look completely wrong

Please open an issue / discussion in the SC repository, as the switch to compose multiplatform was done upstream. Things like the _changed quest setting_ background will be improved later, but I don't yet want to invest much work in compose stuff as I first want to be (relatively) sure upstream is stable here.
Btw you also can't change quest order in the compose settings.

**Reply by mcliquid** [reply]
> I think SC translations are pulled only before SC version update.

So we just have to wait for the next SC version until the translations are available in Weblate?

> Please open an issue / discussion in the SC repository, as the switch to compose multiplatform was done upstream.

Does this mean that it's better to keep the feet still until the switch is complete?

**Reply by Helium314** [reply]
> So we just have to wait for the next SC version until the translations are available in Weblate?

Yes, in the first (alpha?) of 59.0 translations will be available if present in [POEditor](https://poeditor.com/join/project/IE4GC127Ki)

> Does this mean that it's better to keep the feet still until the switch is complete?

I think you can provide feedback already, I don't think the settings UI will change much until the next release.

**Comment by mcliquid** [comment]
@westnordost Before I do any unnecessary work, could you briefly confirm that it makes sense to create tickets for this in StreetComplete? Thank you!

**Reply by westnordost** [reply]
I don't understand what you find is wrong. In any case, yes, it is different now, I redesigned it a bit.

And the font is different? AFAIK the font is the same as before.

**Reply by mcliquid** [reply]
At least the spacing and font sizes look different for me, but never mind.

**Reply by westnordost** [reply]
Well, the upper part of the settings screen now indeed looks a bit more... busy. 

I redesigned that part because:

Settings screen design has never been part of Material design guidelines. Basically, the particular way the settings have been layouted came from a certain (official Android) dependency that we have been using. There is no replacement for that for Compose, one could only reconstruct the UI in order to have it look the same. But I took that as an opportunity to review how the settings were laid out and also to compare it with iOS (settings) design guidelines (they **do** have guidelines for that).

So what I changed, in the end, is to make the placement of the texts more consistent. This is how it is in iOS, too:

Left is always the title of the setting, right always the currently selected value. Below is an explanatory text (when necessary). Before, for some settings, the currently selected value was displayed below (where the explanatory text goes), and for some others, on the right.
The clear separation allowed me to add some (a bit longer) explanation texts for some settings that needed some explanation, IMO. But of course, that makes this screen a bit busier.
